src/wallet/wallet.cpp

```diff
@@ -1606,13 +1626,12 @@ void CWalletTx::GetAmounts(list<COutputEntry>& listReceived,
         // Only need to handle txouts if AT LEAST one of these is true:
         //   1) they debit from us (sent)
         //   2) the output is to us (received)
-        if (nDebit > 0)
+        if (fFromMe)
         {
             // Don't report 'change' txouts
             if (pwallet->IsChange(txout))
                 continue;
-        }
-        else if (!(fIsMine & filter))
+        } if (nDebit == 0 && !(fIsMine & filter))
```

Comment:
break this line in two pls, it's confusing

Reply:
indeed :) I've fixed this (squashed with previous commit)